feat(livestock-animal-set-join): animal set join documentation (#29)
Pure-Farming/Documentation: docs/resource-types/livestock/animal-set-join.md

---
title: Animal Set Join
menu_order: 1
post_status: publish
post_excerpt: Livestock - Animal Set Joins
taxonomy:
  category:
    - api
    - data
  post_tag:
    - api
---

# Resource Type: Livestock - Animal Set Joins

- [URLs](#urls)
- [Response Structure](#response-structure)

---

## URLs

> Get all animal set join that you have access to

Reviewer: I think it is joins. So add a `s` please

```
GET /data/livestock/animal-set-joins
```

Get a specific animal set join

```
GET /data/livestock/animal-set-joins/{animalSetJoinId}
```

Get all animal set joins associated with a holding

```
GET /data/holdings/{holdingId}/livestock/animal-set-joins
```

Get a single animal set join associated with a holding

```
GET /data/holdings/{holdingId}/livestock/animal-set-joins/{animalSetJoinId}
```

---

## Response Structure

A call to the animal set join endpoints returns the following fields.

```json
{
  "@self": "string",
  "location": {...},
  "meta": {...},

>   "resourceType": "/livestock/animal-set",

Reviewer: This should probably be `/livestock/animal-set-join`?

Author: oh, right! good eye. Thanks Markus

  "contemporaryGroup": "string",
  "eventDateTime": "2023-01-03T11:00:00Z",
  "id": "string",
  "remark": "string",
  "reponsible": "string",
  "traitLabel": {...},
  "animal": {...},
}
```

| Response Item          | Description                                                                                                                 | Data Type                                          |
| ---------------------- | --------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------- |

> | **Self**               | A link to this specific animal set                                                                                          | URI                                                |

Reviewer: ```suggestion
| **Self**               | A link to this specific animal set                                                                                          | URI                                                |
```
| **Self**               | A link to this specific animal set  join                                                                                        | URI                                                |

> | **Location**           | An identifier for the location of the animal set                                                                            | [Identifier](/resource-types/common.md#identifier) |

Reviewer: ```suggestion
| **Location**           | An identifier for the location of the animal set join                                                                           | [Identifier](/resource-types/common.md#identifier) |
```

| **Meta**               | Meta data for the resource                                                                                                  | [Metadata](/resource-types/common.md#metadata)     |
| **Resource Type**      | The fixed discriminator for the animal set join resource type.<br/>Value: /livestock/animal-set -join                       | String                                             |
| **Contemporary Group** | For manually recorded events, record any contemporary group code that would affect statistical analysis                     | String                                             |

> | **Event Date Time**    | A particular point in the progression of time                                                                               | Date/Time                                          |

Reviewer: This is clearly not what we want in the dev hub docs. :D

How about the date and time when the animal joined the set?

```

Get a single animal set join event associated with a holding

```
GET /data/holdings/{holdingId}/livestock/animal-set-joins/{animalSetJoinId}
```

---

## Response Structure

A call to the animal set join event endpoints returns the following fields.

```json
{
  "@self": "string",
  "location": {...},
  "meta": {...},
  "resourceType": "/livestock/animal-set-join",
  "contemporaryGroup": "string",
  "eventDateTime": "2023-01-03T11:00:00Z",
  "id": "string",
  "remark": "string",
  "reponsible": "string",
  "traitLabel": {...},
  "animal": {...},
  "animalSetId": "string"
}
```

| Response Item          | Description                                                                                                | Data Type                                          |
| ---------------------- | ---------------------------------------------------------------------------------------------------------- | -------------------------------------------------- |
| **Self**               | A link to this specific animal set join event                                                              | URI                                                |

> | **Location**           | An identifier for the location of the animal set join event                                                | [Identifier](/resource-types/common.md#identifier) |

Reviewer 2: ```suggestion
| **Location**           | An identifier for the location where the animal set join event occurred                      | [Identifier](/resource-types/common.md#identifier) |
```

| **Meta**               | Meta data for the resource                                                                                 | [Metadata](/resource-types/common.md#metadata)     |
| **Resource Type**      | The fixed discriminator for the animal set join event resource type.<br/>Value: /livestock/animal-set-join | String                                             |
| **Contemporary Group** | A contemporary group code that would affect statistical analysis                                           | String                                             |

> | **Event Date Time**    | The time the event occurred                                                                                | Datetime                                           |

Reviewer 2: ```suggestion
| **Event Date Time** | The date and time the event occurred | Datetime |
```
